src/Native/System.Native/pal_mount.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_mount.h"
+#include "pal_utilities.h"
+#include <assert.h>
+#include <string.h>
+#include <errno.h>
+
+// Check if we should use getmntinfo or /proc/mounts
+#if HAVE_MNTINFO
+#include <sys/mount.h>
+#else
+#include <sys/statfs.h>
+#include <mntent.h>
+#define STRING_BUFFER_SIZE 8192
+#endif
+
+#if HAVE_MNTINFO
+
+int GetMountInfo(MountPointFound onFound)
+{
+    // getmntinfo returns pointers to OS-internal structs, so we don't need to worry about free'ing the object
+    struct statfs* mounts = nullptr;
+    int count = getmntinfo(&mounts, 0);
+    for (int32_t i = 0; i < count; i++)
+    {
+        int32_t nameLength = strlen(mounts[i].f_mntonname);
+        onFound(mounts[i].f_mntonname, nameLength);
```

Comment:
It looks to me as though everywhere we call onFound we call strlen first to get get nameLength. Should we just not pass nameLength and let Marshal.PtrToStringAnsi find the terminator?

Also, in the last round, there was something similar where we actually had the length in some cases but not in others and we opted to pass -1 there. If we're going to strlen here, we should probably strlen there too.